addons/pos_coupon/static/src/js/coupon.js

```diff
@@ -264,6 +264,9 @@ odoo.define('pos_coupon.pos', function (require) {
                         this.rewardsContainer = rewardsContainer;
                         // Send a signal that the rewardsContainer are updated.
                         this.trigger('rewards-updated');
+                        for (let reward of this.rewardsContainer.getUnawarded()) {
+                            Gui.showNotification(reward.status.reason);
+                        }
```

Comment:
Can you remove this?

Reply:
In case of any problem with adding a coupon, we don't show any message. You suggested I add it here.

Reply:
Can you remove it, please?

Reply:
Create a different pull request that improves the "notifications" in pos coupon.

Reply:
I removed it, and I'll create new PR for it.
Thanks,